src/scripts/main.js

```diff
@@ -1,3 +1,44 @@
 'use strict';
 
-// write code here
+const thead = document.querySelectorAll('thead tr th');
+const tbody = document.querySelector('tbody');
+const tbodyArray = Array.from(tbody.querySelectorAll('tr'));
+const sortStates = [];
+
+thead.forEach((item) => {
+  sortStates[item] = 'asc';
+
+  item.addEventListener('click', (e) => {
+    const columnIndex = [...thead].indexOf(e.target);
+
+    sortStates[columnIndex] =
+      sortStates[columnIndex] === 'asc' ? 'desc' : 'asc';
```

Comment:
The `sortStates` array is being accessed with `columnIndex`, which is correct. However, ensure that `sortStates` is initialized properly with indices corresponding to column indices.